health/polling_test.go

```diff
@@ -42,24 +42,24 @@ func TestPollingChecker(t *testing.T) {
 	// StateUnhealthy (HTTP error)
 	connection := make(fakeConnChan)
 	close(connection)
-	err := checker.New(ctx, connection, tracker).Close()
```

Comment:
The call to `New` spawns a goroutine that launches a probe. But we'd then immediately call `Close()`, which cancels the prober's context. Both the prober goroutine and this test goroutine are running concurrently, so it's a race as to whether the initial probe can complete before the context is cancelled.